### DISCUSSION TITLE
kitty will not start under ubuntu 20.04 under virtualbox

### DISCUSSION THREAD

**Discussion by pjfarleyiii** [kovidgoyal/kitty#4241] [Q&A]
I run ubuntu 20.04 under virtualbox on a Windows 10 host.  Trying to run kitty, I get the GLFW error reported elsewhere:

`Failed to create GLFW temp window!`

The solution for which seems to be "Install at least v3.3 of OpenGL for kitty to use".  I would gladly install the necessary OpenGL version, but need some guidance on what packages that means I need to install.

glxinfo reports this for the OpenGL settings on the ubuntu system running under virtualbox:

OpenGL vendor string: VMware, Inc.
OpenGL renderer string: SVGA3D; build: RELEASE;  LLVM;
OpenGL version string: 2.1 Mesa 21.0.3
OpenGL shading language version string: 1.20
OpenGL extensions:
OpenGL ES profile version string: OpenGL ES 2.0 Mesa 21.0.3
OpenGL ES profile shading language version string: OpenGL ES GLSL ES 1.0.16
OpenGL ES profile extensions:

Is there an environment variable or config file setting that would let kitty run in this setup without any further software installed?

I also see that the kitty version supplied with ubuntu 20.04 is 0.15.0, is there a repository from which I could use apt install to upgrade that to a newer version?

Any help or advice appreciated.

Peter

**Comment by page-down** [comment]
As far as I can remember, VirtualBox does not have OpenGL 3.3 support.

Try the following environment variables.

```shell
LIBGL_ALWAYS_SOFTWARE=true GALLIUM_DRIVER=llvmpipe kitty
```

https://docs.mesa3d.org/drivers/llvmpipe.html
https://docs.mesa3d.org/drivers/openswr.html

**Reply by santiagoraigoso** [reply]
Hi, is there any way to make it permanent with just one configuration? I had the same problem but kitty doesn't show any errors and just a completely black window appears, although the commands work

**Comment by SatouKuzuma1** [comment]
@page-down  when I reboot virtual box, kitty does not open. I have to rerun the command again and only I can launch kitty from the terminal. Is there any way to set that permanent  ?

**Reply by page-down** [reply]
Searching for the following keywords may be helpful to you.

how to set environment variables on linux
how to create a .desktop entry file
how to create a command alias

Since you're already using macOS, have you considered running kitty on macOS and then using SSH to connect to the VM?

**Comment by SatouKuzuma1** [comment]
I haven't consider that option. What would be the difference ?

**Reply by page-down** [reply]
kitty can run natively under macOS and will be able to use GPU rendering for better performance, plus it can also bind shortcuts directly. No more troubles with VirtualBox.
It depends on what you want to do. If you need an isolated environment to run kitty then it is not suitable.

**Comment by SatouKuzuma1** [comment]
That sounds pretty good! I have never done the ssh connection to a VM so I’ll have to search for it and see how I can do that.
